python/ray/serve/tests/test_cli.py

```diff
@@ -222,5 +223,85 @@ def test_info(ray_start_stop):
     )
 
 
+def parrot(request):
+    return request.query_params["sound"]
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+def test_run(ray_start_stop):
+    # Deploys valid config file and import path via serve run
+
+    # Deploy via config file
+    config_file_name = os.path.join(
+        os.path.dirname(__file__), "test_config_files", "two_deployments.yaml"
+    )
+    p = subprocess.Popen(["serve", "run", "-c", config_file_name])
+    time.sleep(4)
+
+    assert requests.get("http://localhost:8000/shallow").text == "Hello shallow world!"
+    assert requests.get("http://localhost:8000/one").text == "2"
+
+    p.send_signal(signal.SIGINT)  # Equivalent to ctrl-C
+    time.sleep(4)
+
+    with pytest.raises(requests.exceptions.ConnectionError):
+        requests.get("http://localhost:8000/shallow")
+    with pytest.raises(requests.exceptions.ConnectionError):
+        requests.get("http://localhost:8000/one")
+
```

Comment:
don't use `time.sleep`, use `ray.test_utils.wait_for_condition` instead to avoid introducing a flaky test.

Reply:
Switched to `wait_for_condition`.